File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/ui/signup/SignUpService.java
package com.zancheema.android.pantry.ui.signup;

import com.zancheema.android.pantry.ui.signup.dto.SignUpPayload;
import com.zancheema.android.pantry.ui.signup.dto.SignUpResponseBody;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface SignUpService {
    @POST("auth/signup")
    Call<SignUpResponseBody> signup(@Body SignUpPayload payload);
}

File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/ui/signup/SignUpService.kt
package com.zancheema.android.pantry.ui.signup

import com.zancheema.android.pantry.ui.signup.dto.SignUpPayload
import com.zancheema.android.pantry.ui.signup.dto.SignUpResult
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface SignUpService {
    @POST("auth/signup")
    suspend fun signup(@Body payload: SignUpPayload): Response<SignUpResult>
}